src/commons/http.rs

```diff
@@ -2,45 +2,22 @@
 
 use actix_web::web::Bytes;
 use actix_web::Error;
-use hyper::body::to_bytes;
-use hyper::Uri;
+use actix_web::client::Client;
 use log::*;
-use std::str::FromStr;
 
-pub type HyperClient = hyper::Client<
-    hyper_timeout::TimeoutConnector<
-        hyper::client::HttpConnector<hyper::client::connect::dns::GaiResolver>,
-    >,
-    hyper::Body,
->;
-
-pub async fn get_file(client: &HyperClient, url: &str) -> Result<Bytes, Error> {
+pub async fn get_file(client: &Client, url: &str) -> Result<Bytes, Error> {
     debug!("Fetching image from url {}", url);
-    let uri = Uri::from_str(url).map_err(|e| {
-        let error_str = format!("{}", e).replace("\"", "\\\"");
-        error!("Error parsing URI: {}", error_str);
-        actix_web::error::ErrorInternalServerError(e)
-    })?;
-    let response = client.get(uri).await.map_err(move |e| {
-        if let Some(cause) = e.into_cause() {
-            if cause.is::<std::io::Error>() {
-                if let Some(timeout) = cause.downcast_ref::<std::io::Error>() {
-                    if timeout.kind() == std::io::ErrorKind::TimedOut {
-                        error!("Request for image {} timed out", url);
-                        return actix_web::error::ErrorServiceUnavailable(cause);
-                    }
-                }
-            }
-            let error_str = format!("{}", cause).replace("\"", "\\\"");
-            error!("Error getting http file: {}: {}", url, error_str);
-            actix_web::error::ErrorInternalServerError(cause)
-        } else {
-            actix_web::error::ErrorInternalServerError("Unknown error...")
-        }
-    })?;
+    let mut response = client.get(url)
+        .send()
+        .await
+        .map_err(move |e| {
+            let error_str = format!("{}", e).replace("\"", "\\\"");
+            error!("Error to send request: {}", error_str);
+            actix_web::error::InternalError::new(String::from("Error to send request"), actix_http::http::StatusCode::INTERNAL_SERVER_ERROR)
+        })?;
     let status = response.status();
-    if response.status().is_success() {
-        to_bytes(response.into_body()).await.map_err(move |e| {
+    if status.is_success() {
+        response.body().await.map_err(move |e| {
```

Comment:
We need to set the `limit` from the body as it is quite low. We could add a configuration for that 
https://docs.rs/awc/2.0.0-alpha.1/awc/struct.MessageBody.html#method.limit